SRVCOM-1249: Add the Serverless 1.15.0 release notes (#32700)
openshift/openshift-docs: modules/serverless-rn-1-15-0.adoc

// Module included in the following assemblies:
//
// * serverless/release-notes.adoc

[id="serverless-rn-1-15-0_{context}"]
= Release Notes for Red Hat {ServerlessProductName} 1.15.0

[id="new-features-1.15.0_{context}"]
== New features

* {ServerlessProductName} now uses Knative Serving 0.21.0.
* {ServerlessProductName} now uses Knative Eventing 0.21.0.
* {ServerlessProductName} now uses Kourier 0.21.0.
* {ServerlessProductName} now uses Knative `kn` CLI 0.21.0.
* {ServerlessProductName} now uses Knative Kafka 0.21.0.

> * The `serving.knative.dev/visibility` label has been removed. Use `networking.knative.dev/visibility` instead.

Reviewer: Could you add this to __IMPORTANT__ note like this? 

```
[IMPORTANT]
====
* The `serving.knative.dev/visibility` label has been removed. Use `networking.knative.dev/visibility` instead.
====
```
It is not a new feature but users have to take action if they use the old label.

Reviewer: And it would be better to add an explanation about the label. We should add the line like this?:

```
The `serving.knative.dev/visibility` label, which is for creating a private service, has been removed. Use `networking.knative.dev/visibility` label instead if you used.
```

Author: @nak3 Thank you, good suggestions. Implemented:
* https://deploy-preview-32700--osdocs.netlify.app/openshift-enterprise/latest/serverless/serverless-release-notes.html#serverless-rn-1-15-0_serverless-release-notes

Additionally, I've changed the section of the main (non-Release Notes) documentation to account for the new label:
* https://deploy-preview-32700--osdocs.netlify.app/openshift-enterprise/latest/serverless/networking/serverless-ossm-custom-domains.html#knative-service-cluster-local_serverless-ossm-custom-domains

Reviewer: Thank you! And sorry my explanation was bad. I think we can remove `* The serving.knative.dev/visibility label, used for creating private services, has been removed. Use networking.knative.dev/visibility instead.`  in "New features" but just add it only in important note.

- Remove (L#16)
```
* The `serving.knative.dev/visibility` label, used for creating private services, has been removed. Use `networking.knative.dev/visibility` instead.
```

- Add (L#18-21)
```
[IMPORTANT]
====
The `serving.knative.dev/visibility` label, which is for creating a private service, has been removed. Make sure to substitute `serving.knative.dev/visibility` with `networking.knative.dev/visibility` where necessary.
====
```

What do you think?

Author: @nak3 Your explanation was good, and I understood that the whole text needed to go inside the IMPORTANT note. I wanted to do it the same way.

Unfortunately, the AsciiDoc markup language does not allow to have a bullet list item that consists of an IMPORTANT note only. It requires some text to be before the IMPORTANT note. That's why I had to split this note into "before the note" and "inside the note" parts.

If you know how to better split the text between "before the note" and "inside the note" parts, let me know.

(There is also an option where the IMPORTANT note appears after the bullet point list, but it breaks the structure and does not look logical.)

@abrennan89 Can you think of a better way to incorporate Kenjiro's idea?

Author: @nak3 I think it's also good that the part inside the IMPORTANT note is specifically about the most "dangerous" case, _where the reader is already using the old label and needs to take action to change it, otherwise his setup will stop working_. It urges the reader to take the necessary action.
The part before IMPORTANT is informational and does not need to be emphasized so much.

Reviewer 2: I think it's OK to just add it as a note and not a bullet point, but I'd tweak it slightly and put:

"The `serving.knative.dev/visibility` label, that was previously used to create private services, has now been deprecated. You must update any existing services to use the `networking.knative.dev/visibility` label instead."

Author: @abrennan89 Used your suggestion Ashleigh, thanks!

// Module included in the following assemblies:
//
// * serverless/release-notes.adoc

[id="serverless-rn-1-15-0_{context}"]
= Release Notes for Red Hat {ServerlessProductName} 1.15.0

[id="new-features-1.15.0_{context}"]
== New features

* {ServerlessProductName} now uses Knative Serving 0.21.0.
* {ServerlessProductName} now uses Knative Eventing 0.21.0.
* {ServerlessProductName} now uses Kourier 0.21.0.
* {ServerlessProductName} now uses Knative `kn` CLI 0.21.0.
* {ServerlessProductName} now uses Knative Kafka 0.21.1.

[IMPORTANT]
====

> The `serving.knative.dev/visibility` label, that was previously used to create private services, has been deprecated. You must update existing services to use the `networking.knative.dev/visibility` label instead.

Reviewer 3: ```suggestion
The `serving.knative.dev/visibility` label, which was previously used to create private services, is deprecated. You must update existing services to use the `networking.knative.dev/visibility` label instead.
```

> ====

Reviewer 4: /cc @devguyio

> ====

Author: @matzew Thanks, added.

> ====

Author: Removed the note because Ahmed has fixed the issue.

// Module included in the following assemblies:
//
// * serverless/release-notes.adoc

[id="serverless-rn-1-15-0_{context}"]
= Release Notes for Red Hat {ServerlessProductName} 1.15.0

[id="new-features-1.15.0_{context}"]
== New features

* {ServerlessProductName} now uses Knative Serving 0.21.0.
* {ServerlessProductName} now uses Knative Eventing 0.21.0.
* {ServerlessProductName} now uses Kourier 0.21.0.
* {ServerlessProductName} now uses Knative `kn` CLI 0.21.0.
* {ServerlessProductName} now uses Knative Kafka 0.21.0.

> * The `serving.knative.dev/visibility` label, used for creating private services, has been removed. Use `networking.knative.dev/visibility` instead.
> +
> [IMPORTANT]
> ====
> Make sure to substitute `serving.knative.dev/visibility` with `networking.knative.dev/visibility` where necessary.
> ====

Reviewer 2: ```suggestion
[IMPORTANT]
====
The `serving.knative.dev/visibility` label, that was previously used to create private services, has been deprecated. You must update existing services to use the `networking.knative.dev/visibility` label instead.
====
```

// Module included in the following assemblies:
//
// * serverless/release-notes.adoc

> [id="serverless-rn-1-15-0_{context}"]
> 
> = Release Notes for Red Hat {ServerlessProductName} 1.15.0

Reviewer 2: ```suggestion
[id="serverless-rn-1-15-0_{context}"]
= Release Notes for Red Hat {ServerlessProductName} 1.15.0
```
remove extra line space?

[id="new-features-1.15.0_{context}"]
== New features

* {ServerlessProductName} now uses Knative Serving 0.21.0.
* {ServerlessProductName} now uses Knative Eventing 0.21.0.
* {ServerlessProductName} now uses Kourier 0.21.0.
* {ServerlessProductName} now uses Knative `kn` CLI 0.21.0.

> * {ServerlessProductName} now uses knative-kafka 0.21.0.

Reviewer 2: ```suggestion
* {ServerlessProductName} now uses Knative Kafka 0.21.0.
```

> // Should we include Knative `net-istio` 0.21.0?

Reviewer 2: No, we've never previously mentioned it. Maybe double check with @markusthoemmes or QE folks.

Reviewer 5: Nope, please don't mention net-istio at all.

Author: @markusthoemmes Thanks Markus.

+
[IMPORTANT]
====
Only the `v1beta1` version of the APIs for `KafkaChannel` and `KafkaSource` objects on {ServerlessProductName} are supported. Do not use the `v1alpha1` version of these APIs, as this version is now deprecated.

> ====

Reviewer 2: can remove this, it's in the 1.14.0 relnotes and main docs now

// Module included in the following assemblies:
//
// * serverless/release-notes.adoc

[id="serverless-rn-1-15-0_{context}"]
= Release Notes for Red Hat {ServerlessProductName} 1.15.0

[id="new-features-1.15.0_{context}"]
== New features

* {ServerlessProductName} now uses Knative Serving 0.21.0.
* {ServerlessProductName} now uses Knative Eventing 0.21.0.
* {ServerlessProductName} now uses Kourier 0.21.0.
* {ServerlessProductName} now uses Knative `kn` CLI 0.21.0.

> * {ServerlessProductName} now uses Knative Kafka 0.21.0.

Reviewer 4: we are actually using `0.21.1` version: https://github.com/openshift-knative/serverless-operator/blob/release-1.15/olm-catalog/serverless-operator/manifests/serverless-operator.clusterserviceversion.yaml#L551
